recipes/llvm-core/all/conanfile.py

```diff
@@ -52,13 +53,75 @@
     "XCore"
 }
 
+def components_from_dotfile(dotfile):
```

Comment:
Why is it required only now? I saw your feature request in Conan client for Dot (Graphviz), still, this is a bit complex to maintain, so I would avoid it if possible. Boost is a good example of how opaque the recipe becomes after receiving customizations.

Reply:
See [this comment](https://github.com/conan-io/conan-center-index/pull/22997#discussion_r1725129187) from #22997 for the context behind this decision.

In summary, parsing CMake Graphviz output was how the original (Conan 1.x) LLVM recipe generated component dependencies - I removed this in favour of parsing CMake output, however in the above referenced comment it was suggested that Graphviz was the "least-worst" option, so I've added it back in here.

Reply:
Could you please add more detailed comments in the recipe, including the method description, and what is happening in the key steps? As I commented before, it not only increases the recipe's complexity but also makes its maintenance harder. More information will be welcome to understand it in future PRs.